### ISSUE TITLE
version.upgrade()

### ISSUE THREAD

**Issue by Rowingwolf** [nwestfall/BlazorDB#11]
I would recommend you add code to fire an event when version.upgrade() fires.  That allows you to make manual schema changes or modify data when the version number changes.

**Comment by tfrank7415** [comment]
@Rowingwolf I don't think it makes sense to add an event that will fire when version.upgrade() fires. Dexie handles the manual schema changes and data modification for us if we use an upgrader function. See below image. More information can be found at https://dexie.org/docs/Version/Version.upgrade()

![Sample](https://user-images.githubusercontent.com/44216318/209047104-fc9fcc91-d3e5-4502-85ca-4121b766e21e.PNG)

**Comment by tfrank7415** [comment]
I forked the repo to work on this issue. I'm trying to develop a solution that will allow us to collect the necessary information in c# then build the upgrader function (db.version(x).stores().upgrade()) in blazorDB.js. I'm experimenting with allowing the user to add a StorageSchemaUpgrades object via DbStore options in AddBlazorDb() in program.cs. See below.

![Capture](https://user-images.githubusercontent.com/44216318/209047762-c4d8aef5-602a-4b68-860a-f45ccbcfefff.PNG)

I am having success with performing one schema update via the upgrader function per version update. I have not attempted to use multiple upgrader function for one version update yet. Please provide feedback. I am actively researching for the best way to build the upgrader function. @Rowingwolf @nwestfall
